### PR TITLE
fix(event/create): ensure .location matches .locations[]

### PR DESCRIPTION
**Pull request by w-le**

### CODE REVIEW

File: src/models/event.cr [hunk 1]
```diff
@@ -108,11 +108,14 @@ module Office365
...
       if typeof(location) == String
+        # Use the provided location
         @location = Location.new(display_name: location)
-      elsif !locs.empty?
-        @location = Location.new(display_name: locs.compact_map(&.display_name).join("; "))
+        @locations = [@location.not_nil!]
+      else
+        # Generate location based on invited resources
+        @locations = locations = @attendees.compact_map { |a| Location.new(display_name: a.name) if a.type == AttendeeType::Resource }
+        @location = Location.new(display_name: locations.join("; "))
```
Comment (stakach) [inline src/models/event.cr]: this should be `@location = Location.new(display_name: locations.compact_map(&.display_name).join("; "))`

**Review by stakach — CHANGES_REQUESTED** [review]

File: src/models/event.cr (hunk 1 above)
Comment (w-le) [reply src/models/event.cr]: oh yeah totally

Comment (w-le) [reply src/models/event.cr]: actually I think would be fine for `location` to be JUST the first location, as that's exactly what the MS examples do.
`@location = locations.first`

File: src/models/event.cr [hunk 2]
```diff
@@ -108,11 +108,14 @@ module Office365
...
-      @locations = locs = @attendees.compact_map { |a| Location.new(display_name: a.name) if a.type == AttendeeType::Resource }
       if typeof(location) == String
+        # Use the provided location
         @location = Location.new(display_name: location)
-      elsif !locs.empty?
-        @location = Location.new(display_name: locs.compact_map(&.display_name).join("; "))
+        @locations = [@location.not_nil!]
+      else
+        # Generate location based on invited resources
+        @locations = locations = @attendees.compact_map { |a| Location.new(display_name: a.name) if a.type == AttendeeType::Resource }
```
Comment (caspiano) [inline src/models/event.cr]:
```suggestion
        @locations = locations = @attendees.compact_map { |a| Location.new(display_name: a.name) if a.type.resource? }
```

File: src/models/event.cr [hunk 3]
```diff
@@ -108,11 +108,14 @@ module Office365
 
       # As outlined in the example request
       # https://docs.microsoft.com/en-us/graph/api/user-post-events?view=graph-rest-1.0&tabs=http#request-1
-      @locations = locs = @attendees.compact_map { |a| Location.new(display_name: a.name) if a.type == AttendeeType::Resource }
       if typeof(location) == String
```
Comment (caspiano) [inline src/models/event.cr]:
```suggestion
      if location.is_a? String
```

**Pull request merged**